File: src/main/java/com/quotes/api/entities/Quote.java
package com.quotes.api.entities;

import java.time.LocalDate;
import java.time.ZoneId;
import java.util.Date;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "quotes")
public class Quote {
    @Id
    private String id;
    private String author;
    private String work;
    private String message;
    private String[] tags;
    private Date dateCreated;

    public String getAuthor() {
        return author;
    }

    public String getId() { return id; }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getWork() {
        return work;
    }

    public void setWork(String work) {
        this.work = work;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String[] getTags() {
        return tags;
    }

    public void setTags(String[] tags) {
        this.tags = tags;
    }

    public Date getDateCreated() {
        return dateCreated;
    }

    public void setDateCreated() {
        this.dateCreated = Date.from(LocalDate.now().atStartOfDay().atZone(ZoneId.systemDefault()).toInstant());
    }
}

File: internal/domain/quotes/quote.go
package quotes

import "time"

type Quote struct {
	QuoteID     int64     `json:"quote_id"`
	Author      string    `json:"author"`
	Work        string    `json:"work"`
	Phrase      string    `json:"phrase"`
	Tags        string    `json:"Tags"`
	DateCreated time.Time `json:"date_created"`
}

func (d Quote) GetDateCreatedFormatted() string {
	return d.DateCreated.Format("Enero 02, 2006")
}
